Packs/AzureLogAnalytics/Integrations/AzureLogAnalytics/AzureLogAnalytics.py

```diff
@@ -94,32 +90,34 @@ def http_request(self, method, url_suffix=None, full_url=None, params=None,
 ''' INTEGRATION HELPER METHODS '''
 
 
-def format_query_table(table):
+def validate_params(refresh_token, managed_identities_client_id, client_credentials, enc_key, self_deployed,
+                    certificate_thumbprint, private_key):
+    if not refresh_token:
+        raise DemistoException('Token / Tenant ID must be provided.')
+    if not managed_identities_client_id:
+        if client_credentials and not enc_key:
+            raise DemistoException("Client Secret must be provided for client credentials flow.")
+        elif not self_deployed and not enc_key:
+            raise DemistoException('Key must be provided. For further information see '
+                                    'https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication')  # noqa: E501
+        elif not enc_key and not (certificate_thumbprint and private_key):
+            raise DemistoException('Key or Certificate Thumbprint and Private Key must be provided.')
+
+
+def format_query_table(table: dict[str, Any]):
     name = table.get('name')
-    columns = [column.get('name') for column in table.get('columns')]
-    rows = table.get('rows')
+    columns = [column.get('name') for column in table.get('columns', [])]
+    rows = table.get('rows', [])
     data = [
         dict(zip(columns, row)) for row in rows
     ]
 
     return name, columns, data
 
 
-def query_output_to_readable(tables):
-    readable_output = '## Query Results\n'
-    tables_markdown = [tableToMarkdown(name=name,
-                                       t=data,
-                                       headers=columns,
-                                       headerTransform=pascalToSpace,
-                                       removeNull=True) for name, columns, data in tables]
-    readable_output += '\n'.join(tables_markdown)
-
-    return readable_output
-
-
-def flatten_saved_search_object(saved_search_obj):
-    ret = saved_search_obj.get('properties')
-    ret['id'] = saved_search_obj.get('id').split('/')[-1]
+def flatten_saved_search_object(saved_search_obj: dict[str, Any]) -> dict[str, Any]:
+    ret: dict = saved_search_obj.get('properties', {})
+    ret['id'] = saved_search_obj.get('id', '').split('/')[-1]
```

Comment:
Are we sure there will be / in the id and that it will not be empty? ''.split('/') == [] which will fail on index error if we try to access [-1] of it.

Reply:
i didn't get index error,
`''.split('/')`
['']
`''.split('/')[-1]`
''